### DISCUSSION TITLE
GitHub Actions Workflows - deployment approval timeouts

### DISCUSSION THREAD

**Discussion by larryclaman** [community/community#5673] [Actions]
When using GitHub Actions Workflows with deployment approvals, I'm looking for the ability to time-out a workflow if no one approves it within a specified period of time.

As a comparison, when I've used Azure DevOps Pipelines, I can add a manual approval to a pipeline, and set that approval to time out after X minutes. Eg if no one approves it after X minutes, the pipeline is canceled (as if someone clicked 'not approved')

However, when using GitHub Actions Workflows with environment approvals, the [documentation](https://docs.github.com/en/actions/managing-workflow-runs/reviewing-deployments#about-required-reviews-in-workflows) says _"If a job is not approved within 30 days, the workflow run will be automatically canceled_".  But it doesn't appear that you can customize the 30 day timeout to something more useful; eg, hours or minutes.  So unless you manually clean it up, you can end up with a lot of workflows "stuck" in the waiting-for-approval state.

Please add the capability to customize the 30 day timeout to something user specified.

**Comment by HariSekhon** [comment]
This is especially important because unapproved deployments can backlog and prevent new ones from being able to run

**Reply by Orrimp** [reply]
With a self-hosted runner 10 PRs per Week, 30 Days is a bit excessive :D 
For us one week waiting would be enough. Therefore please make it configurarble

**Comment by apoclyps** [comment]
The ability to configure a timeout for deployments awaiting approval would be hugely beneficial. I work on a team that typically deploys in the range of 10-20 a day from Monday to Friday across a range of services and the number of Github Action Workflows awaiting approval can be significant.

A configurable timeout would be ideal for the required approval steps to avoid waiting 30 days for a cleanup operation to happen

**Comment by nemmeviu** [comment]
+1

**Comment by melendezd** [comment]
Being able to customize the timeout for deployment approvals would be very useful. We are trying to migrate our deployments to GitHub Actions and not having this feature complicates things.

**Comment by AFMiziara** [comment]
+1

**Comment by micd** [comment]
+1

**Comment by tfglasswing** [comment]
+2

**Comment by nnamacha** [comment]
+1

**Comment by grilla99** [comment]
+1

**Comment by mikebollandajw** [comment]
Copilot told me the answer to this was 

```timeout-minutes: 240 # 4 hours```

That appeared to be a lie, is there no other option?

**Comment by priyank-vaghela** [comment]
+1

**Comment by donald-berning** [comment]
We have the opposite problem most are noting.  Our build/deploy process can span over 30 days until finally in production.  A configurable solution at the workflow action run level (longer for production candidates, shorter for development only) would be ideal.

**Comment by blinchi** [comment]
+1

**Comment by juanPablo06** [comment]
Used the same feature in ADO. I was very surprised that this is not an option in GHA.

**Comment by balaram-gilead** [comment]
+1

**Comment by RotemBirman** [comment]
+1

**Comment by Googlom** [comment]
+1

<details>
  <summary>spoiler</summary>
    never sell your product to Microsoft
</details>

**Comment by Skywalker427** [comment]
5 years later...
+1

**Comment by fardarter** [comment]
+1
